### DISCUSSION TITLE
Large field in form data is truncated on server

### DISCUSSION THREAD

**Discussion by gurumaxi** [sveltejs/kit#10804] [Q&A]
I've implemented a form in sveltekit, and in one of its fields (hidden field) I store a large base64 encoded image, which on form submit will be sent to the server. The issue is that this field is truncated at approx. 1.048.000 characters on the server. How can I prevent this?
- I am not using use:enhance on the form
- I am using the vercel adapter, but the issue is reproducible locally and in production
- The payload size is 1.1MB (Max body size of the vercel adapter is 4.5MB)
- When trying to send the same content of the field to a sveltekit api endpoint, it just works fine, no issues there
- In the browsers network tab I can see that in the payload the field has the full correct value, and it is not truncated

So there has to be some sort of issue related to forms.

**Comment by Shiinoya** [comment]
Did you figure this out? I had this issue and finally figured out a way to make it work.

Instead of using a hidden field, I submit the image file directly to my `FormActions` as a `Blob` and once in SvelteKit's backend I do all the conversions needed (also using `jimp`).

It seems to be a limitation on HTML form inputs and how much data they can send. I think.

**Comment by ljazzy** [comment]
relevant!

**Comment by leoj3n** [comment]
@ljazzy This may be a helpful read...

https://vercel.com/docs/storage/vercel-blob/client-upload

I have not had to worry about this myself so not sure.

**Comment by Shiinoya** [comment]
In my previous comment I erroneously blamed HTML for my lack of doc reading abilities. Turns out the node adapter comes with a [default upload limit](https://kit.svelte.dev/docs/adapter-node#environment-variables-bodysizelimit). Removing or adjusting this limit was the proper solution.

I did however maintain the implementation of using a `Blob`.

**Reply by leoj3n** [reply]
Good to know, thanks for sharing. Seems like the adapter or SvelteKit should throw some kind of notification when that limit is hit if not already doing so. Does it not?

**Reply by Shiinoya** [reply]
It throws, in my opinion, an unhelpful error. Can't remember exactly, but something along the lines of "| is not a valid JSON entry". It was a few months ago, but I can replicate the error if that'd be helpful.

**Reply by leoj3n** [reply]
Probably could suggest a PR to add an explanation to the Error that the user may want to increase `BODY_SIZE_LIMIT` here:

https://github.com/sveltejs/kit/blob/main/packages/kit/src/exports/node/index.js#L39C9-L39C16

**Reply by Shiinoya** [reply]
That's not a bad idea. Although I do see there has been changes in the code since I last ran into the issue. I'll have to test things out to make sure it's still relevant.

**Comment by wwmike** [comment]
Same for me almost a year later.
